packages/api-server/src/methods/validator.ts

```diff
@@ -273,11 +273,62 @@ export function verifyBlockParameter(
     return undefined;
   }
 
-  const err = verifyHexNumber(blockParameter, index);
-  if (err) {
-    return err.padContext("blockParameter block number");
+  if (typeof blockParameter === "object") {
+    const err = verifyBlockSpecifier(blockParameter, index);
+    if (err) {
+      return err.padContext("blockSpecifier");
+    }
+  } else {
+    const err = verifyHexNumber(blockParameter, index);
+    if (err) {
+      return err.padContext("blockParameter block number");
+    }
+  }
+
+  return undefined;
+}
+
+export function verifyBlockSpecifier(
+  blockSpecifier: BlockSpecifier,
+  index: number
+) {
+  if (typeof blockSpecifier !== "object") {
+    return invalidParamsError(index, `blockSpecifier must be an object`);
+  }
+
+  if (blockSpecifier.blockHash == null && blockSpecifier.blockNumber == null) {
+    return invalidParamsError(
+      index,
+      "blockSpecifier has no blockHash and blockNumber"
+    );
   }
 
+  if (blockSpecifier.blockHash != null && blockSpecifier.blockNumber != null) {
+    return invalidParamsError(
+      index,
+      "blockHash and blockNumber can not exits at same time"
+    );
+  }
+
+  if (blockSpecifier.requireCanonical === true) {
```

Comment:
Why true is disallowed?

Reply:
we don't have canonical blocks, so disallowed.

to be compatible with Ethereum,  requireCanonical should be kept according to the specs of eip1898. Throw error when un-compatible is more clear for users.

Reply:
Oh, this is the opposite of what I understood. As far as I know, Godwoken only has canonical blocks.

Reply:
you are right,  I misunderstood it as uncle block

Reply:
updated